### PR TITLE
feat: add unit tests and improve code readability

### PR DESCRIPTION
**Pull request by CauaneAndrade**
This pull request includes the following changes:

- Added unit tests "test_app_run_fails_folder_exists" and "test_app_run_success" to ensure proper functionality of the App::run method
- Improved code commenting and removed unnecessary imports and mut variables to increase code readability and maintainability

Please review the changes and let me know if there are any issues or further improvements that should be made before merging this pull request. Thank you!

### CODE REVIEW

**Comment by CauaneAndrade** [comment]
@M4urici0GM could you please take a look at this PR? I would appreciate your feedback and suggestions on the changes made. :)

File: src/app.rs [hunk 1]
```diff
@@ -67,26 +62,54 @@ impl App {
...
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use tempfile::tempdir;
+    use crate::args::AppArguments;
+
+    #[test]
+    fn test_app_run_success() {
+        let temp_dir = tempdir().unwrap();
```
Comment (M4urici0GM) [inline src/app.rs:89]: Cool, i think that's it!!
There's a pattern that i like to follow, it's called AAA Test pattern, where you split the initialization, the execution and the assertion into smaller 'sections'
[Here](https://medium.com/@pablodarde/o-padr%C3%A3o-triple-a-arrange-act-assert-741e2a94cf88) is a good article about it.
Once you start using it, you never stop, its a lot easier for understanding what the test does.
(Not a required change, just a suggestion)

**Review by M4urici0GM — APPROVED** [review]
LGTM!!
There's a few things we can change, we could do a Enum based error, instead of using strings for error everywhere, but it can be in another PR.

```rust
pub enum AppError {
 ProjectAlreadyExist,
 FileAlreadyExists,
 FileNotFound,
 Other(Box<dyn Error>),
}
```
you would need to change your Result<T> type to something like
```rust
type Result<T> = std::result::Result<T, AppError>;
```

then we would impl the Display trait, so we can display the correct messages based on the type itself, and on the main file, something like
```rust
...
match app.run() {
  Ok(_) => println!("Success"),
  Err(err) => println!("Error when creating the project: {}", err),
 }
 ```
 what do you think? for the next PR

**Comment by CauaneAndrade** [comment]
Hey @M4urici0GM , thanks for the suggestion on the AAA Test pattern - I'll definitely take a look at that article and consider implementing it in future PRs.

I really like the idea of using an Enum-based error approach, it will make the code more readable and maintainable. I'll definitely work on creating an AppError Enum and implementing the Display trait in the next PR. Thanks for taking the time to review and for providing valuable feedback, it's greatly appreciated.

**Pull request merged**